File: explore/libwnck-3-dev/gi/gjs/prototype/mousepad-move-to-workspace/mousepad-current-workspace.js
#!/usr/bin/env gjs

imports.gi.versions.Gtk = '3.0';
imports.gi.versions.Gdk = '3.0';

//print(imports.gi.versions.Gdk)

const Gdk = imports.gi.Gdk;
const Gtk = imports.gi.Gtk;
const GLib = imports.gi.GLib;
const Wnck = imports.gi.Wnck;


Gtk.init(null);


let on_window_opened = function (screen, window) {

	/*
	print("application_name:", window.get_application().get_name());
	print("window_name:", window.get_application().get_name());
	print("window_class:", window.get_class_group_name());
	print("window_xid:", window.get_xid());
	*/

	if (window.get_class_group_name() === 'Mousepad') {
		//https://lazka.github.io/pgi-docs/index.html#Wnck-3.0/classes/Workspace.html#Wnck.Workspace.get_number
		print("workspace:", window.get_workspace().get_number());
	}

	loop.quit();
}


let screen = Wnck.Screen.get_default() // https://lazka.github.io/pgi-docs/index.html#Wnck-3.0/classes/Screen.html#Wnck.Screen.get_default
screen.connect('window-opened', on_window_opened) // https://lazka.github.io/pgi-docs/index.html#Wnck-3.0/classes/Screen.html#Wnck.Screen.signals.window_opened


let loop = GLib.MainLoop.new(null, false);
loop.run();

File: explore/libwnck-3-dev/gi/python/prototype/mousepad-move-to-workspace/mousepad-current-workspace.py
#!/usr/bin/env python3

import gi
gi.require_version('Gtk', '3.0')
gi.require_version('Wnck', '3.0')
from gi.repository import GLib, Gtk, Wnck


def on_window_opened(screen, window):

	if window.get_class_group_name() == 'Mousepad':
		# https://lazka.github.io/pgi-docs/index.html#Wnck-3.0/classes/Workspace.html#Wnck.Workspace.get_number
		print("workspace: {workspace}".format(workspace=window.get_workspace().get_number()));


	loop.quit() # https://lazka.github.io/pgi-docs/index.html#GLib-2.0/classes/MainLoop.html#GLib.MainLoop.quit


if __name__ == '__main__':

	# https://lazka.github.io/pgi-docs/index.html#Gtk-3.0/functions.html#Gtk.init
	# https://lazka.github.io/pgi-docs/index.html#Gdk-3.0/functions.html#Gdk.init
	Gtk.init(None);

	screen = Wnck.Screen.get_default() # https://lazka.github.io/pgi-docs/index.html#Wnck-3.0/classes/Screen.html#Wnck.Screen.get_default
	screen.connect('window-opened', on_window_opened) # https://lazka.github.io/pgi-docs/index.html#Wnck-3.0/classes/Screen.html#Wnck.Screen.signals.window_opened

	loop = GLib.MainLoop(None) # https://lazka.github.io/pgi-docs/index.html#GLib-2.0/classes/MainLoop.html#GLib.MainLoop
	loop.run()
